### PR TITLE
Track password edits:

### LINKED ISSUE #1454: Add password changed field to userprofile metadata
It is useful to let an API consumer know when a user's password was last changed.  Let us add a `last_password_edit` field to the user profile metadata that communicates it.

### Context

The Kaznet API needs to know if a user changed their password via ona.io so that the Kaznet mobile app can anticipate and prevent ODK authentication issues by forcing users to provide their new password.

### LINKED ISSUE #1467: Error Thrown When Creating Account
The following error is thrown when creating an account. Issue affects [v1.14.6](https://github.com/onaio/onadata/tree/v1.14.6):

```
ERROR 2018-09-03 07:40:23,026 exception 31485 140183367087936 Internal Server Error: /api/v1/profiles
Traceback (most recent call last):
  File "/home/onadata/.virtualenvs/onadata/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/home/onadata/.virtualenvs/onadata/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/onadata/.virtualenvs/onadata/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/onadata/.virtualenvs/onadata/local/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/onadata/.virtualenvs/onadata/local/lib/python2.7/site-packages/rest_framework/viewsets.py", line 103, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/onadata/.virtualenvs/onadata/local/lib/python2.7/site-packages/rest_framework/views.py", line 483, in dispatch
    response = self.handle_exception(exc)
  File "/home/onadata/.virtualenvs/onadata/local/lib/python2.7/site-packages/rest_framework/views.py", line 443, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/onadata/.virtualenvs/onadata/local/lib/python2.7/site-packages/rest_framework/views.py", line 480, in dispatch
    response = handler(request, *args, **kwargs)
  File "./onadata/apps/api/viewsets/user_profile_viewset.py", line 151, in create
    data['metadata'] = {'last_password_edit': timezone.now().isoformat()}
  File "/home/onadata/.virtualenvs/onadata/local/lib/python2.7/site-packages/django/http/request.py", line 434, in __setitem__
    self._assert_mutable()
  File "/home/onadata/.virtualenvs/onadata/local/lib/python2.7/site-packages/django/http/request.py", line 431, in _assert_mutable
    raise AttributeError("This QueryDict instance is immutable")
AttributeError: This QueryDict instance is immutable
```

### PR DESCRIPTION
**Pull request by lincmba**
Introduce last_password_edit field in the metadata json field
 Upon creation of an account add the last_password_edit time
 Upon password change update last_password_edit
 fixes #1454 
 fixes #1467

### CODE REVIEW

File: onadata/apps/api/viewsets/user_profile_viewset.py [hunk 1]
```diff
@@ -154,10 +155,14 @@ def change_password(self, request, user):  # pylint: disable=W0613
         current_password = request.data.get('current_password', None)
         new_password = request.data.get('new_password', None)
 
+        metadata = user_profile.metadata or {}
```
Comment (ukanga) [inline onadata/apps/api/viewsets/user_profile_viewset.py]: Why not have the the newly introduced `metadata` variable within the `if new_password` block? Is it used anywhere else outside the block?

File: onadata/libs/serializers/user_profile_serializer.py [hunk 2]
```diff
@@ -253,14 +255,16 @@ def create(self, validated_data):
 
         created_by = request.user
         created_by = None if created_by.is_anonymous() else created_by
+        metadata['last_password_edit'] = timezone.now().isoformat()
```
Comment (ukanga) [inline onadata/libs/serializers/user_profile_serializer.py]: This is being updated everytime there is profile update, not everytime there was a password change.

File: onadata/apps/api/viewsets/user_profile_viewset.py (hunk 1 above)
Comment (ukanga) [inline onadata/apps/api/viewsets/user_profile_viewset.py]: Why not have the the newly introduced `metadata` variable within the `if new_password` block? Is it used anywhere else outside the block?

File: onadata/libs/tests/serializers/test_user_profile_serializer.py [hunk 3]
```diff
@@ -75,3 +78,28 @@ def test_get_temp_token_recreates_if_expired(self):
         is_expired = expired(temp_token.created)
 
         self.assertFalse(is_expired)
+
+
+class TestUserProfileSerializer(TestAbstractViewSet):
+
+    def test_metadata_view_for_owner_only(self):
```
Comment (ukanga) [inline onadata/libs/tests/serializers/test_user_profile_serializer.py]: How is this test related to the issue being addressed by this pull-request on password edits?

File: onadata/libs/serializers/user_profile_serializer.py (hunk 2 above)
Comment (lincmba) [reply onadata/libs/serializers/user_profile_serializer.py]: Actually it's every time an account is created not every time there's an account update.

File: onadata/libs/tests/serializers/test_user_profile_serializer.py (hunk 3 above)
Comment (lincmba) [reply onadata/libs/tests/serializers/test_user_profile_serializer.py]: The field inclusion had been taken care of in different tests. This test is to ensure that the last_password_edit field can only be viewed by the account owner, for security reasons.
